### PR TITLE
Feature/revamp documentation

### PR DESCRIPTION
**Pull request by VictorGosse**
Breaking changes:
- `Badge` component doesn't get the `exponent` prop anymore as this can be done easily by extending the component.
- `ButtonGroup` component now use its own theme values instead of the one of `InputGroup`. That said, the default values are the same.
- `InputCalendar` component now has a better design for the selected day. Associated theme values has been updated.
- `Menu` component has been totally revamped. It now follows the compound pattern instead of adding the elements as props. Associated theme has been updated.
- `Radio` component is now a controlled component.
- `Option` and `OptionPanel` components have been removed.

### CODE REVIEW

File: packages/atti-components/src/components/Collapsible/Readme.mdx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+`
+
+const App = () => {
+  const [isOpen, setIsOpen] = useState(false)
+
+  return (
+    <>
+      <Button onClick={() => setIsOpen(prevIsOpen => !prevIsOpen)}>Collapse</Button>
+      <Collapsible open={isOpen}>
+        <Content>Coucou</Content>
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Collapsible/Readme.mdx]: Maybe keep the text content in english ?

File: packages/atti-components/src/components/Container/Readme.mdx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+---
+
+Container is the basic component to use to wrap the main content of your page. It'll center it horizontally.
+
+## Example
+
+```js
+<Container>
+  <Text>This is some content</Text>
+  <Text>No special layout is applyed to this content as this is a raw container</Text>
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Container/Readme.mdx]: applied

File: packages/atti-components/src/components/Grid/Readme.mdx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+path: '/grid'
+name: 'Grid'
+title: 'Grid'
+---
+
+Grid component works together with the Cell component. It's reponsible for displaying correctly the Cells based on the size of the viewport.
+
+Basically there are 12 columns for xlDesktopand and desktop. And there are 6 columns for tablet and mobile.
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Grid/Readme.mdx]: whitespace missing

File: packages/atti-components/src/components/Input/Readme.mdx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+---
+
+Let the user enter and edit text. This is the basic element of forms.
+
+## Examples
+
+Normal Input type text :
+
+```js
+<Input id="inputTestTxt" name="inputTestTxt" placeholder="Input text" type="text" />
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Input/Readme.mdx]: type isn't `text` by default ?

File: packages/atti-components/src/components/Input/index.js [hunk 5]
```diff
@@ -42,27 +40,24 @@ const Input = ({
...
 Input.propTypes = {
-  /**
-   * The classes of the input field.
-   */
-  className: PropTypes.string,
   /**
    * The id of the input field.
    */
-  id: PropTypes.string.isRequired,
+  id: PropTypes.string,
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Input/index.js:60]: Why remove the requirement ? I reckon `ids` should be mandatory for all input elements no ?

File: packages/atti-components/src/components/InputCalendar/Readme.mdx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+```js
+<InputCalendar onChange={date => console.log(date)} />
+```
+
+Controlled component:
+
+```js
+const Example = () => {
+  const [date, setDate] = useState(new Date())
+  return <InputCalendar value={date} onChange={date => setDate({ date })} />
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/InputCalendar/Readme.mdx]: Why give an object to the setter ?

File: packages/atti-components/src/components/InputCalendar/styles/StyledDay.js [hunk 7]
```diff
@@ -1,24 +1,42 @@
...
+    align-items: center;
+    justify-content: center;
+    border-radius: 50%;
+    ${({ isSameMonth, theme }) =>
+      !isSameMonth ? `color: ${theme.components.inputCalendar.colors.notSameMonth};` : ''}
+    ${({ isSunday, theme }) =>
+      isSunday ? `color: ${theme.components.inputCalendar.colors.sunday};` : ''}
+    ${({ isToday, theme }) =>
+      isToday ? `color: ${theme.components.inputCalendar.colors.today};` : ''}
+    ${({ isSelected, theme }) =>
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/InputCalendar/styles/StyledDay.js]: You can probably simplify these arrow functions and the duplication of `theme.components.inputCalendar.colors` by using a function

File: packages/atti-components/src/components/InputPopdown/Readme.mdx [hunk 8]
```diff
@@ -0,0 +1,56 @@
+---
+path: '/inputpopdown'
+name: 'InputPopdown'
+title: 'InputPopdown'
+---
+
+InputPopdown will display a component when you click onn another component.
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/InputPopdown/Readme.mdx]: on

File: packages/atti-components/src/components/InputPopdown/Readme.mdx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+```js
+const Example = () => {
+  const [value, setValue] = useState(50)
+  return (
+    <>
+      <InputPopdown enhancer={() => <ProgressBar value={value} />}>
+        <InputGroup>
+          <Button onClick={() => setValue(prevState => prevState + 1)}>More</Button>
+          <Input onChange={value => setState({ value })} value={value} />
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/InputPopdown/Readme.mdx]: Why not use `setValue` ?

File: packages/atti-components/src/components/InputPopdown/Readme.mdx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+          <Button onClick={() => setValue(prevState => prevState - 1)}>Less</Button>
+        </InputGroup>
+      </InputPopdown>
+      <Text>
+        Here we enhance the stardard input with a loader and More/Less buttons for more sexyness !
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/InputPopdown/Readme.mdx]: standard

File: packages/atti-components/src/components/Menu/components/BurgerMenu/index.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { Holder, Line } from './styles'
+
+const BurgerMenu = props => {
+  const { opened, updateOpened } = props
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Menu/components/BurgerMenu/index.js]: Why not do this in the function's arguments ?

File: packages/atti-components/src/components/Menu/components/BurgerMenu/styles/Holder.js [hunk 12]
```diff
@@ -0,0 +1,50 @@
+import styled from 'styled-components'
+
+import Line from './Line'
+
+const Holder = styled.div`
+  display: none;
+  width: ${({ theme }) => theme.components.menu.burgerMenu.spaces.width};
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Menu/components/BurgerMenu/styles/Holder.js]: You will not like it, but you will gain quite some lines and perfs by using only one arrow function

File: packages/atti-components/src/components/Menu/components/BurgerMenu/styles/Holder.js [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+  height: ${({ theme }) => theme.components.menu.burgerMenu.spaces.height};
+  transform: translateX(
+    ${({ opened, theme }) => (opened ? theme.components.menu.vertical.spaces.width : '0')}
+  );
+  transition: transform 0.3s ease;
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Menu/components/BurgerMenu/styles/Holder.js]: Shouldn't we use some theme for the transitions ? So we have a consistent animation time across our components

File: packages/atti-components/src/components/Menu/components/BurgerMenu/styles/Line.js [hunk 14]
```diff
@@ -0,0 +1,12 @@
+import styled from 'styled-components'
+
+const Line = styled.div`
+  position: absolute;
+  height: ${({ theme }) => theme.components.menu.burgerMenu.line.spaces.height};
+  width: ${({ theme }) => theme.components.menu.burgerMenu.spaces.width};
+  background-color: ${({ theme }) => theme.components.menu.burgerMenu.line.colors.background};
+  transition: transform 0.3s ease-in-out;
+  border-radius: ${({ theme }) => theme.components.menu.burgerMenu.line.borderRadius};
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Menu/components/BurgerMenu/styles/Line.js]: can use only one arrow function here too

File: packages/atti-components/src/components/Menu/components/ItemGroup/index.js [hunk 15]
```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import ItemLink from '../Item/styles/ItemLink'
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Menu/components/ItemGroup/index.js]: You made an index and you're not using it ? :p

File: packages/atti-components/src/components/Menu/components/ItemGroup/index.js [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+import { Arrow, ItemsContainer, Li, Title } from './styles'
+
+import MenuContext from '../../context/MenuContext'
+
+const ItemGroup = ({ children, title }) => {
+  return (
+    <MenuContext.Consumer>
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Menu/components/ItemGroup/index.js]: Using a hook will reduce verbosity

File: packages/atti-components/src/components/Menu/components/ItemGroup/styles/ItemsContainer.js [hunk 17]
```diff
@@ -0,0 +1,12 @@
+import styled from 'styled-components'
+
+import Collapsible from 'components/Collapsible'
+import ItemLink from '../../Item/styles/ItemLink'
+
+const ItemsContainer = styled(Collapsible)`
+  ${ItemLink} {
+    padding-left: ${({ theme }) => theme.spaces.s4} !important;
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Menu/components/ItemGroup/styles/ItemsContainer.js]: You shouldn't need to use `! important`. When you add styles to a styled-component from its parent, it priority is higher than the styled-component's style itself.
If you still need the `!important` it means there's a pattern problem in the way you organize your components

File: packages/atti-components/src/components/Menu/index.js [hunk 18]
```diff
@@ -1,62 +1,67 @@
...
+const Menu = ({ children, isResponsive, isVertical }) => {
+  const [isOpened, setIsOpened] = useState(false)
+  const [subMenuOpened, setSubMenuOpened] = useState('')
+
+  const updateOpened = () => {
+    setIsOpened(prevIsOpened => !prevIsOpened)
+  }
+
+  return (
+    <MenuContext.Provider
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Menu/index.js]: You can encapsulate the Context.Provider in a Component to add flexibility to the component

File: packages/atti-components/src/components/Menu/index.js [hunk 19]
```diff
@@ -1,62 +1,67 @@
...
-  variance: PropTypes.oneOf(['basic', 'withButtons', 'withLiftedTabs']),
+  isVertical: PropTypes.bool,
 }
 
-/** @component */
-export default styled(Menu)``
+Menu.Item = Item
+Menu.ItemGroup = ItemGroup
+
+export default Menu
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Menu/index.js]: We cannot extend the Menu as a styled-component any more ? Why ?

File: packages/atti-components/src/components/Menu/styles/MenuHolder.js [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+import { Arrow, ItemsContainer } from '../components/ItemGroup/styles'
+
+const MenuHolder = styled.div`
+  height: ${({ theme }) => theme.components.menu.spaces.height};
+  background-color: ${({ theme }) => theme.components.menu.colors.background};
+
+  /* HORIZONTAL MENU */
+  ${({ isVertical, theme }) =>
+    !isVertical &&
+    css`
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Menu/styles/MenuHolder.js]: don't need this, you're already inside a css`` method

File: packages/atti-components/src/components/Menu/styles/MenuHolder.js [hunk 21]
```diff
@@ -0,0 +1,59 @@
+import styled, { css } from 'styled-components'
+
+import verticalMenu from './verticalMenu'
+
+import { ItemLink } from '../components/Item/styles'
+import { Arrow, ItemsContainer } from '../components/ItemGroup/styles'
+
+const MenuHolder = styled.div`
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/Menu/styles/MenuHolder.js:8]: You can reduce heavily the code base of this file by centralizing the arrow functions

File: packages/atti-components/src/components/ProgressBar/index.js [hunk 22]
```diff
@@ -22,6 +21,8 @@ const ProgressBar = ({
   label,
   leftLabel,
   rightLabel,
+
```
Comment (alexandre-lelain) [inline packages/atti-components/src/components/ProgressBar/index.js]: Felt like adding extra line ? :p

**Comment by alexandre-lelain** [comment]
Awesome work 👍

File: packages/atti-components/src/components/Collapsible/Readme.mdx (hunk 1 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Collapsible/Readme.mdx]: Fixed

File: packages/atti-components/src/components/Container/Readme.mdx (hunk 2 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Container/Readme.mdx]: Fixed

File: packages/atti-components/src/components/Input/Readme.mdx (hunk 4 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Input/Readme.mdx]: You're right

File: packages/atti-components/src/components/Input/index.js (hunk 5 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Input/index.js:60]: No it's not. It's useful to link it with a label for example but not mandatory.

File: packages/atti-components/src/components/InputPopdown/Readme.mdx (hunk 8 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/InputPopdown/Readme.mdx]: Fixed

File: packages/atti-components/src/components/InputPopdown/Readme.mdx (hunk 9 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/InputPopdown/Readme.mdx]: It's an error. Fixed

File: packages/atti-components/src/components/InputPopdown/Readme.mdx (hunk 10 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/InputPopdown/Readme.mdx]: Fixed

File: packages/atti-components/src/components/Menu/components/BurgerMenu/index.js (hunk 11 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Menu/components/BurgerMenu/index.js]: Done

File: packages/atti-components/src/components/Menu/components/ItemGroup/index.js (hunk 15 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Menu/components/ItemGroup/index.js]: I've totally changed this part. Should be better now :)

File: packages/atti-components/src/components/Menu/components/ItemGroup/index.js (hunk 16 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Menu/components/ItemGroup/index.js]: Should be better now :)

File: packages/atti-components/src/components/Menu/components/ItemGroup/styles/ItemsContainer.js (hunk 17 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Menu/components/ItemGroup/styles/ItemsContainer.js]: Should be better. Was indeed a wrong pattern at some point.

File: packages/atti-components/src/components/Menu/styles/MenuHolder.js (hunk 20 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Menu/styles/MenuHolder.js]: Fixed

File: packages/atti-components/src/components/Grid/Readme.mdx (hunk 3 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Grid/Readme.mdx]: Fixed

File: packages/atti-components/src/components/InputCalendar/Readme.mdx (hunk 6 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/InputCalendar/Readme.mdx]: Fixed

File: packages/atti-components/src/components/Menu/components/BurgerMenu/styles/Holder.js (hunk 12 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Menu/components/BurgerMenu/styles/Holder.js]: You're right, much more better now.

File: packages/atti-components/src/components/Menu/components/BurgerMenu/styles/Line.js (hunk 14 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Menu/components/BurgerMenu/styles/Line.js]: 👍 Done

File: packages/atti-components/src/components/Menu/styles/MenuHolder.js (hunk 21 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Menu/styles/MenuHolder.js:8]: Done

File: packages/atti-components/src/components/ProgressBar/index.js (hunk 22 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/ProgressBar/index.js]: Fixed

File: packages/atti-components/src/components/InputCalendar/styles/StyledDay.js (hunk 7 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/InputCalendar/styles/StyledDay.js]: Done

File: packages/atti-components/src/components/Menu/components/BurgerMenu/styles/Holder.js (hunk 13 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Menu/components/BurgerMenu/styles/Holder.js]: Good catch, done :)

File: packages/atti-components/src/components/Menu/index.js (hunk 18 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Menu/index.js]: Good idea

File: packages/atti-components/src/components/Menu/index.js (hunk 19 above)
Comment (VictorGosse) [reply packages/atti-components/src/components/Menu/index.js]: Should be good now.

**Review by alexandre-lelain — DISMISSED** [review]

**Review by atrakeur — APPROVED** [review]

**Pull request merged**